### PR TITLE
140444579: Add missing server down test case api-scheduler integration test.

### PR DESCRIPTION
**Pull request by kanekoh**
Will apply the code refactor to all other integration test in a separate story.

### CODE REVIEW

File: src/integration/integration_api_scheduler_test.go [hunk 1]
```diff
@@ -70,6 +70,34 @@ var _ = Describe("Integration_Api_Scheduler", func() {
...
+				BeforeEach(func() {
+					stopScheduler(schedulerProcess)
+
+					policyStr = readPolicyFromFile("fakePolicyWithSchedule.json")
+					resp, err := attachPolicy(appId, policyStr)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(resp.StatusCode).To(Equal(http.StatusInternalServerError))
+					resp.Body.Close()
+
+					schedulerProcess = startScheduler()
```
Comment (fraenkel) [inline src/integration/integration_api_scheduler_test.go]: Why is this here?
It should be in an AfterEach.

File: src/integration/integration_api_scheduler_test.go [hunk 2]
```diff
@@ -94,6 +122,34 @@ var _ = Describe("Integration_Api_Scheduler", func() {
...
+				BeforeEach(func() {
+					stopScheduler(schedulerProcess)
+
+					policyStr = readPolicyFromFile("fakePolicyWithoutSchedule.json")
+					resp, err := attachPolicy(appId, policyStr)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(resp.StatusCode).To(Equal(http.StatusCreated))
+					resp.Body.Close()
+
+					schedulerProcess = startScheduler()
```
Comment (fraenkel) [inline src/integration/integration_api_scheduler_test.go]: AfterEach

I would restructure the tests slightly.
Pull the "Scheduler is down" up, and really we only need 1 test since with or without schedule should affect the outcome but I would do it with a schedule since that would be something the scheduler participates in.

File: src/integration/integration_api_scheduler_test.go [hunk 3]
```diff
@@ -168,6 +224,30 @@ var _ = Describe("Integration_Api_Scheduler", func() {
 				Expect(resp.StatusCode).To(Equal(http.StatusNotFound))
 				resp.Body.Close()
 			})
+
+			Context("Scheduler is down", func() {
```
Comment (fraenkel) [inline src/integration/integration_api_scheduler_test.go]: Move this.

File: src/integration/integration_api_scheduler_test.go (hunk 1 above)
Comment (kanekoh) [inline src/integration/integration_api_scheduler_test.go]: The reason is to check if there are no schedules in scheduler or not with REST API.
Is it enough to check API server respond status is Internal Server Error when API server received a policy with schedules while schedule is down?

Comment (fraenkel) [inline src/integration/integration_api_scheduler_test.go]: Sorry, my comment was the startScheduler.
The context is when the scheduler is down. Therefore we should never start the scheduler.
It should be started in the AfterEach, i.e., after the test is complete not before.

Comment (kanekoh) [inline src/integration/integration_api_scheduler_test.go]: I was not clear in my explanation before. The startScheduler is part of the test case to verify the state of the data. Maybe better to have it in the "It" section. But still I understand we should have another start scheduler in the afterEach in case something goes wrong in the middle.

File: src/integration/integration_api_scheduler_test.go (hunk 2 above)
Comment (kanekoh) [inline src/integration/integration_api_scheduler_test.go]: Yes understand your point about having only with schedule.
The point we were trying to make by having a test without schedule was to verify the scheduler doesn't participate when there is no schedule.

File: src/integration/integration_api_scheduler_test.go (hunk 1 above)
Comment (fraenkel) [inline src/integration/integration_api_scheduler_test.go]: You don't need to verify the scheduler since the API server will tell us the answer.

File: src/integration/integration_api_scheduler_test.go (hunk 2 above)
Comment (fraenkel) [inline src/integration/integration_api_scheduler_test.go]: If we know it is down, then we also know the data cannot change.

**Comment by kanekoh** [comment]
ok, will fix them.
